pippy/PipelineSchedule.py

```diff
@@ -719,31 +719,30 @@ def backward_stage_local_index(step):
                     bwd_mb_index == self._n_microbatches - 1
                 )
                 logger.debug(
-                    f"{self.rank}: {step=}, {fwd_stage.stage_index=}, {bwd_stage.stage_index=}, {fwd_mb_index=}, {bwd_mb_index=}"
+                    f"Rank {self.rank}: {step=}, {fwd_stage.stage_index=}, {bwd_stage.stage_index=}, {fwd_mb_index=}, {bwd_mb_index=}"
                 )
                 with record_function(f"1F1B {step}"):
-                    ops = fwd_stage.get_fwd_recv_ops()
+                    ops.extend(fwd_stage.get_fwd_recv_ops())
                     ops.extend(bwd_stage.get_bwd_recv_ops())
-                    works = sorted_batch_isend_irecv(ops)
-                    for work in works.values():
+                    if ops:
+                        work = dist.batch_isend_irecv(ops).pop()
                         work.wait()
+                        sends_to_wait.append(work)
```

Comment:
nit: if we are batching sends *and recvs* together, then this is no need to push `work` to `sends_to_wait`. The latter is for sends only.

Reply:
good point!